uniffi_bindgen/src/bindings/kotlin/templates/ObjectTemplate.kt

```diff
@@ -298,6 +307,17 @@ open class {{ impl_class_name }}: Disposable, AutoCloseable, {{ interface_name }
             {{ impl_class_name }}({% call kt::to_ffi_call(cons) %})
         {% endfor %}
     }
+    {% else if is_error %}
+    companion object ErrorHandler : UniffiRustCallStatusErrorHandler<{{ impl_class_name }}> {
+        override fun lift(error_buf: RustBuffer.ByValue): {{ impl_class_name }} {
+            // Due to some mismatches in the ffi converter mechanisms, errors are a RustBuffer.
```

Comment:
I wonder if we could get rid of that restriction at this point -- that can be another follow up.